modules/tinymce/src/plugins/table/test/ts/browser/settings/TableDefaultAttributesWithColGroupsTest.ts

```diff
@@ -0,0 +1,130 @@
+import { ApproxStructure, GeneralSteps, Logger, Pipeline, Waiter } from '@ephox/agar';
+import { UnitTest } from '@ephox/bedrock-client';
+import { TinyApis, TinyLoader, TinyUi } from '@ephox/mcagar';
+import TablePlugin from 'tinymce/plugins/table/Plugin';
+import SilverTheme from 'tinymce/themes/silver/Theme';
+import * as TableTestUtils from '../../module/test/TableTestUtils';
+
+/* This requires a menubar. Cannot migrate yet. */
+UnitTest.asynctest('browser.tinymce.plugins.table.TableDefaultAttributesWithColGroupsTest', (success, failure) => {
+  TablePlugin();
+  SilverTheme();
+
+  TinyLoader.setup((editor, onSuccess, onFailure) => {
+    const tinyApis = TinyApis(editor);
+    const tinyUi = TinyUi(editor);
+
+    const createTableChildren = (s, str: ApproxStructure.StringApi, _arr) =>
+      [
+        s.element('colgroup', {
+          children: [
+            s.element('col', {
+              styles: {
+                width: str.contains('%')
+              }
+            }),
+            s.element('col', {
+              styles: {
+                width: str.contains('%')
+              }
+            })
+          ]
+        }),
+        s.element('tbody', {
+          children: [
+            s.element('tr', {
+              children: [
+                s.element('td', {
+                  styles: {
+                    width: str.none()
+                  },
+                  children: [
+                    s.element('br', {})
+                  ]
+                }),
+                s.element('td', {
+                  styles: {
+                    width: str.none()
+                  },
+                  children: [
+                    s.element('br', {})
+                  ]
+                })
+              ]
+            }),
+            s.element('tr', {
+              children: [
+                s.element('td', {
+                  styles: {
+                    width: str.none()
+                  },
+                  children: [
+                    s.element('br', {})
+                  ]
+                }),
+                s.element('td', {
+                  styles: {
+                    width: str.none()
+                  },
+                  children: [
+                    s.element('br', {})
+                  ]
+                })
+              ]
+            })
+          ]
+        })
+      ];
+
+    Pipeline.async({}, [
+      Logger.t('no attributes without setting', GeneralSteps.sequence([
```

Comment:
`Logger` is deprecated, please use the `Log` module instead, as that'll also require linking it to a JIRA.

Reply:
Sure.